CoffeeShopUI_Angular/src/app/components/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,88 @@
+;
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { isSubscription } from 'rxjs/internal/Subscription';
+@Component({
+  selector: 'sign-up',
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.css'],
+})
+export class SignUpComponent {
+  constructor(private fb: FormBuilder, private http: HttpClient) {}
+  isSubmitted = false;
+ 
+  signUpForm = this.fb.group({
+    firstName: ['', Validators.required],
+    lastName: ['', Validators.required],
+    email: ['', [Validators.required, Validators.email]],
+    password: [
+      '',
+      [
+        Validators.required,
+        // Validators.pattern('(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[.,@$!%*#?&^_-]).{8,99}'),
+      ],
+      
+    ],
+    isSubscripToNewsletter: false,
+  });
+  onSubmit(): void {{
+    console.log(
+      'signUpForm form',
+      this.signUpForm.value,
+      this.signUpForm.valid
+    );
+    const headers = new HttpHeaders({
+      'Content-Type': 'application/json',
+      'Content-Length':'<calculated when request is sent>',
+      'User-Agent':'PostmanRuntime/7.33.0',
+      'Accept-Encoding':'gzip, deflate, br',
+      'Connection': 'keep-alive',
+
+    });
+
+    const requestBody = {
+      email: this.signUpForm.get('email')?.value,
+      firstName: this.signUpForm.get('firstName')?.value,
+      lastName: this.signUpForm.get('lastName')?.value,
+      role: this.signUpForm.get('role')?.value,
+      password: this.signUpForm.get('password')?.value,
+      newsLetter: {
+        email: this.signUpForm.get('email')?.value,
+        name: this.signUpForm.get('firstName')?.value + " "+ this.signUpForm.get('lastName')?.value,
+        isActived: this.signUpForm.get('isSubscripToNewsletter')?.value
+      }
+    };
+    
+    this.http
+    .post<any>('https://localhost:7282/api/Auth/RegisterUser', requestBody)
```

Comment:
It's good practice to look into environment specific variables. 
Here, for example, you are using localhost as an endpoint, this is OK for tutorials and practicing, but not good in actual work. 

As you might already know, projects have certain development environments ( e.g. local, development, staging, production ) 

Hint: In angular we have environment specific files: environment.ts, environment.prod.ts, environment.dev.ts. 
Look up ways of having different api endpoints based on what environment need you have.